### DISCUSSION TITLE
Run API and UI test sequentially

### DISCUSSION THREAD

**Discussion by meetveera** [artilleryio/artillery#1614] [Artillery]
Hi @hassy - Good Morning, Hope you are doing well :)

I have a specific question regarding the test which I am doing. I want to perform some UI testing followed by APIs and than again UI  that too sequentially. The way I am doing it that adding my UI Scripts in before and after scenarios, but the problem with that is the metrics does not report the before and after scenarios. So is there any way where I can acheive this and also get data in my metrics? 
Thannks

**Comment by hassy** [comment]
hi @meetveera - at the moment, the only way to do it would be to run 3 different Artillery commands, one after another, e.g.:

```sh
artillery run ui-test1.yaml
artillery run load-test.yml
artillery run ui-test2.yml
```

You can extract the common configuration into a separate file to avoid duplication, and re-use it with the `--config` flag, e.g.:

```sh
artillery run --config common.yml ui-test1.yml
```

More on the `--config` flag in the docs: https://docs.art/guides/guides/command-line#extract-and-reuse-configuration

The ability to pin specific scenarios to phases has come up before, but we don't have an ETA for implementing that at the moment. I'd love to learn more about your use-case.
